packages/odie-client/src/components/message/index.tsx

```diff
@@ -82,73 +88,103 @@ const ChatMessage = ( { message, currentUser }: ChatMessageProps ) => {
-					) }
-					className="odie-chatbox-thinking-icon"
-				/>
-			) : (
-				<strong className="message-header-name">{ botName }</strong>
-			) }
-
-			<div className="message-header-buttons">
-				{ message.content?.length > 600 && (
-					<Button compact borderless onClick={ handleFullscreenToggle }>
+	let messageAvatarHeader;
+	switch ( message.role ) {
+		case 'user':
+			messageAvatarHeader = (
+				<>
+					<Gravatar
+						user={ currentUser }
+						size={ 32 }
+						alt={ _x( 'User profile display picture', 'html alt tag', __i18n_text_domain__ ) }
+					/>
+					<strong className="message-header-name">{ currentUser.display_name }</strong>
+				</>
+			);
+			break;
+		case 'agent':
+			messageAvatarHeader = (
+				<>
+					<img
+						src={ AgentAvatar }
+						alt={ _x( 'Support agent profile picture', 'html alt tag', __i18n_text_domain__ ) }
+						className={ wapuuAvatarClasses }
+					/>
+					<strong className="message-header-name">
+						{ _x( 'WordPress.com Support', 'Support message header', __i18n_text_domain__ ) }
+					</strong>
+				</>
+			);
+			break;
+		case 'bot':
+		default:
+			messageAvatarHeader = (
+				<>
+					<img
+						src={ WapuuAvatar }
+						alt={ sprintf(
+							/* translators: %s is bot name, like Wapuu */
+							_x( '%(botName)s profile picture', 'html alt tag', __i18n_text_domain__ ),
+							botName
+						) }
+						className={ wapuuAvatarClasses }
+					/>
+					{ message.type === 'placeholder' ? (
 						<img
-							src={ isFullscreen ? MinimizeIcon : MaximizeIcon }
+							src={ WapuuThinking }
 							alt={ sprintf(
 								/* translators: %s is bot name, like Wapuu */
 								_x(
-									'Icon to expand or collapse %(botName)s messages',
+									'Loading state, awaiting response from %(botName)s',
 									'html alt tag',
 									__i18n_text_domain__
 								),
 								botName
 							) }
+							className="odie-chatbox-thinking-icon"
 						/>
-					</Button>
-				) }
-			</div>
-		</>
-	);
+					) : (
+						<strong className="message-header-name">{ botName }</strong>
+					) }
 
-	const messageHeader = (
-		<div className={ `message-header ${ isUser ? 'user' : 'bot' }` }>{ messageAvatarHeader }</div>
-	);
+					<div className="message-header-buttons">
+						{ message.content?.length > 600 && (
+							<Button compact borderless onClick={ handleFullscreenToggle }>
+								<img
+									src={ isFullscreen ? MinimizeIcon : MaximizeIcon }
+									alt={ sprintf(
+										/* translators: %s is bot name, like Wapuu */
+										_x(
+											'Icon to expand or collapse %(botName)s messages',
+											'html alt tag',
+											__i18n_text_domain__
+										),
+										botName
+									) }
+								/>
+							</Button>
+						) }
+					</div>
+				</>
+			);
+	}
+
+	let messageHeaderClass = 'message-header';
+	if ( isUser ) {
+		messageHeaderClass += ' user';
+	} else if ( isAgent ) {
+		messageHeaderClass += ' agent';
+	} else {
+		messageHeaderClass += ' bot';
+	}
```

Comment:
`clsx` is designed for this case exactly. And it's already imported in this file.